masu/processor/tasks.py

```diff
@@ -71,23 +71,31 @@ def get_report_files(customer_name,
     # initiate chained async task
     LOG.info('Reports to be processed: %s', str(reports))
     for report_dict in reports:
-        stats = ReportStatsDBAccessor(os.path.basename(report_dict.get('file')))
-        last_start = stats.get_last_started_datetime()
-        last_end = stats.get_last_completed_datetime()
+        file_name = os.path.basename(report_dict.get('file'))
+        stats = ReportStatsDBAccessor(file_name)
+        started_date = stats.get_last_started_datetime()
+        completed_date = stats.get_last_completed_datetime()
         stats.close_session()
 
-        # only queue up a processing job if there's none currently in-progress.
-        if not last_start or not last_end or last_start < last_end:
-            request = {'schema_name': schema_name,
-                       'report_path': report_dict.get('file'),
-                       'compression': report_dict.get('compression')}
-            result = process_report_file.delay(**request)
-            LOG.info('Processing task queued - File: %s, Task ID: %s',
-                     report_dict.get('file'),
-                     str(result))
-        else:
-            LOG.info('Processing task NOT queued. Start time is later than end time! Detail: %s',
-                     str(report_dict))
+        # Skip processing if already in progress.
```

Comment:
Any thought to having this be in a time window or something? I’m concerned we could have a worker die during processing and miss the data. Just something to think about bi suppose worst case is we will process the next update.

Reply:
Yeah maybe if started_date is saved, worker dies before completed timestamp can be saved, then we would never re-process... 
These timestamps are done here:
```
    stats_recorder = ReportStatsDBAccessor(file_name)
    stats_recorder.log_last_started_datetime()
    stats_recorder.commit()
    processor = ReportProcessor(schema_name=schema_name,
                                report_path=report_path,
                                compression=compression)

    processor.process()
    stats_recorder.log_last_completed_datetime()
    stats_recorder.commit()
    stats_recorder.close_session()
```
I think if we remove the commit before processing the database would roll back the change if something bad happened during processing.  @adberglund would this type of change make logging the start/end timestamp be an atomic operation? @blentz, do you remember the reason for adding this commit before processing?  Is the reason still valid now that we know we will never be re-processing the same file outside of a worker dying type situation?

Reply:
Pretty sure it was also for the SNS + Polling case, where it was possible to pick up the same file, so we wanted to log as soon as possible that we had started processing.

Reply:
I don't see it as a blocker, just we may want to open an issue to think about how we want to handle this edge case.

Reply:
Yeah, I don't want to let this hold up this change since we are wasting lots of resources as is today.  But I'll look at adding a simple timeout window around this check.  If started time is an hour ago with no completed time, then just re-process